ocaml/tests/test_clustering.ml

```diff
@@ -313,11 +313,65 @@ let test_assert_pif_prerequisites () =
 let test_assert_pif_prerequisites =
   [ "test_assert_pif_prerequisites", `Quick, test_assert_pif_prerequisites ]
 
+
+(** Test PIF.disallow_unplug is RO when clustering is enabled *)
+let check_disallow_unplug expected_value __context pif msg =
+  Alcotest.(check bool) msg
+    (Db.PIF.get_disallow_unplug ~__context ~self:pif)
+    expected_value
+
+(* Need host and network to make PIF *)
+let make_host_network_pif ~__context =
+  let host = Test_common.make_host ~__context () in
+  let network = Test_common.make_network ~__context () in
+  let pif = Test_common.make_pif ~__context ~network ~host () in
+  (host, network, pif)
+
+(* Test PIF.set_disallow_unplug without cluster/cluster_host objects *)
+let test_disallow_unplug_no_clustering () =
+  let __context = Test_common.make_test_database () in
+  let host,network,pif = make_host_network_pif ~__context in
+
+  (* Test toggling disallow_unplug when disallow_unplug:false by default *)
+  check_disallow_unplug false __context pif
+    "check_disallow_unplug called by test_disallow_unplug_no_clustering when testing default config";
+  Xapi_pif.set_disallow_unplug ~__context ~self:pif ~value:true;
+  check_disallow_unplug true __context pif
+    "check_disallow_unplug called by test_disallow_unplug_no_clustering after setting disallow_unplug";
+
+  (* Test toggling disallow_unplug when initialised to true *)
+  let pif_no_unplug = Test_common.make_pif ~__context ~network ~host ~disallow_unplug:true () in
+  check_disallow_unplug true __context pif_no_unplug
+    "check_disallow_unplug called by test_disallow_unplug_no_clustering when initialising disallow_unplug:true";
+  Xapi_pif.set_disallow_unplug ~__context ~self:pif_no_unplug ~value:false;
+  check_disallow_unplug false __context pif_no_unplug
+    "check_disallow_unplug called by test_disallow_unplug_no_clustering after setting disallow_unplug:false"
+
+let test_disallow_unplug_with_clustering () =
+  let __context = Test_common.make_test_database () in
+  let host,network,pif = make_host_network_pif ~__context in
+  check_disallow_unplug false __context pif
+    "check_disallow_unplug called by test_disallow_unplug_with_clustering to check default config";
+
+  (* PIF.disallow_unplug should become RO upon introduce cluster_host object *)
+  let _ = Test_common.make_cluster_and_cluster_host ~__context ~network ~host () in
+  Alcotest.check_raises
+    "check_disallow_unplug called by test_disallow_unplug_with_clustering after attaching cluster and cluster_host to network"
+    (Api_errors.(Server_error(clustering_enabled_on_network, [Ref.string_of network])))
+    (fun () -> Xapi_pif.set_disallow_unplug ~__context ~self:pif ~value:true)
```

Comment:
You might want to add a second check that setting `disallow_unplug` to false raises as well. In the current implementation it is obvious it will, but the real goal is that we reject setting it to false.